### ISSUE TITLE
import Tensorflow 2.2 models

### ISSUE THREAD

**Issue by cpaterop** [opencv/opencv#17434]
##### System information (version)
<!-- Example
- OpenCV => 4.2
- Operating System / Platform => Windows 64 Bit
- Compiler => python 3.8

##### Detailed description
Hi guys,
i am trying to create the .pbtxt file of my tf model and i folow your tutorial
https://github.com/opencv/opencv/wiki/TensorFlow-text-graphs

i think it works perfectly for tensorflow 1 models but not for 2.2
any ideas?

##### Steps to reproduce

import tensorflow as tf
from tensorflow import keras
from tensorflow.keras.layers import Dense 
import numpy as np

def get_model():
  inputs = keras.Input(shape=(32,))
  outputs = keras.layers.Dense(1)(inputs)
  model = keras.Model(inputs, outputs)
  model.compile(optimizer='adam', loss='mean_squared_error')
  return model

model = get_model()

test_input = np.random.random((128, 32))
test_target = np.random.random((128, 1))
model.fit(test_input, test_target)

# Calling `save('my_model')` creates a SavedModel folder `my_model`.
model.save('my_model')

pb_file = '/content/my_model/saved_model.pb' # use the right path
graph_def = tf.compat.v1.GraphDef()

try:
    with tf.io.gfile.GFile(pb_file, 'rb') as f:
        graph_def.ParseFromString(f.read())
except:
    with tf.gfile.FastGFile(pb_file, 'rb') as f:
        graph_def.ParseFromString(f.read())

# Delete weights
for i in reversed(range(len(graph_def.node))):
    if graph_def.node[i].op == 'Const':
        del graph_def.node[i]

graph_def.library.Clear()

**Comment by dkurt** [comment]
Please add more details what's TensorFlow 2.2 model. It'd be useful to have an example.

Also take a look at SavedModel import examples at https://github.com/opencv/opencv/issues/16582#issuecomment-603819498 and https://github.com/opencv/opencv/issues/16879#issuecomment-603815872

**Comment by cpaterop** [comment]
Hi @dkurt, thanks for the insights
actually i cannot manage to load the .pb in dnn for both the mnist and the mobile net model (although with some manipulation i can produce the frozen graph). 
In order to verify that it is not my installation that causes the problem i also run it in colab/google with the same results (opencv 4.1.2, TF 2.2.0)

of course the same with my own model (attached) that takes (none 20,20,1). i would appreciate it if you had a look

[saved_model.zip](https://github.com/opencv/opencv/files/4718913/saved_model.zip)
 kind regards,
Christos

**Comment by dkurt** [comment]
> opencv 4.1.2

Please try newer version, at least 4.3.0

**Comment by cpaterop** [comment]
actually i just managed to have it work.
save model to .hdf5 and then freeze the graph with one of your scripts:
`
model_hdf5 = 'model.hdf5'
frozen_pb_file = '/content/frozen_graph.pb'

model.save(model_hdf5)

print(' ----FREEZE GRAPH----- ')
frozen_pb_file = '/content/frozen_graph.pb'

from tensorflow.python.framework.convert_to_constants import convert_variables_to_constants_v2
from tensorflow.python.tools import optimize_for_inference_lib
loaded = keras.models.load_model(model_hdf5)
flatten_input=tf.TensorSpec(shape=[None, 20, 20,1], dtype=tf.float32)
f = tf.function(loaded).get_concrete_function(flatten_input)
f2 = convert_variables_to_constants_v2(f)
graph_def = f2.graph.as_graph_def()

# Export frozen graph
with tf.io.gfile.GFile('frozen_graph.pb', 'wb') as f:
   f.write(graph_def.SerializeToString())

print('done')
